s4e-backend/src/test/java/pl/cyfronet/s4e/TestDbHelper.java

```diff
@@ -16,8 +19,14 @@
     private final PRGOverlayRepository prgOverlayRepository;
     private final SldStyleRepository sldStyleRepository;
     private final PropertyRepository propertyRepository;
+    private final ProductCategoryRepository productCategoryRepository;
 
     public void clean() {
+        StreamSupport
+                .stream(productCategoryRepository.findAll().spliterator(), false)
+                .filter(productCategory -> !productCategory.getLabel().contains(ProductCategoryRepository.DEFAULT_CATEGORY_LABEL))
+                .peek(productCategory -> productCategoryRepository.delete(productCategory))
+                .collect(Collectors.toList());
```

Comment:
I don't like the idea in general, yet I don't feel like fighting it. Nevertheless, don't overcomplicate easy things.

Declare a method `void deleteAllByLabelNot(String label)` on the repository, and then:
```java
productCategoryRepository.deleteAllByLabelNot(ProductCategoryRepository.DEFAULT_CATEGORY_LABEL)
```

Reply:
The problem will occur in case of circular dependency while providing @Autowired repository inside of category to